source/MaterialXCore/Interface.cpp

```diff
@@ -355,8 +362,29 @@ OutputPtr Input::getConnectedOutput() const
     return result;
 }
 
+InputPtr Input::getInterface() const
+{
+    const string& interfaceName = getInterfaceName();
+    if (!interfaceName.empty())
+    {
+        ConstNodeGraphPtr graph = getAncestorOfType<NodeGraph>();
+        if (graph && !graph->hasNodeDefString())
+        {
+            return graph->getInput(interfaceName);
+        }
+    }
+    return nullptr;
+}
+
 NodePtr Input::getConnectedNode() const
 {
+    // Traverse through interface names to nodegraph input
+    InputPtr graphInput = getInterface();
+    if (graphInput && (graphInput->hasNodeName() || graphInput->hasNodeGraphString()))
```

Comment:
If this input has an interfacename, then traverse through to that input to find any upstream connection.